lib/publiccloud/ec2.pm

```diff
@@ -114,14 +114,40 @@ sub upload_img {
 
     die("Create key-pair failed") unless ($self->create_keypair($self->prefix . time, 'QA_SSH_KEY.pem'));
 
+    # AMI of image to use for helper VM to create/build the image on CSP.
+    my $helper_ami_id = get_var('PUBLIC_CLOUD_EC2_UPLOAD_AMI');
+
+    # in case AMI for helper VM is not provided in job settings fallback to predefined hash
+    unless (defined($helper_ami_id)) {
+
+        # AMI is region specific also we need to use different AMI's for on-demand/BYOS uploads
+        my $ami_id_hash = {
+            # suse-sles-15-sp2-byos-v20201211-hvm-ssd-x86_64
+            'us-west-1-byos' => 'ami-02b1a98531379c966',
+            # suse-sles-15-sp2-v20201211-hvm-ssd-x86_64
+            'us-west-1' => 'ami-05c558c169cfe8d99',
+            # suse-sles-15-sp2-byos-v20201211-hvm-ssd-x86_64
+            'us-west-2-byos' => 'ami-0b167c7e41ab53cb1',
+            # suse-sles-15-sp2-v20201211-hvm-ssd-x86_64
+            'us-west-2' => 'ami-0174313b5af8423d7',
+            # suse-sles-15-sp2-byos-v20201211-hvm-ssd-x86_64
+            'eu-central-1-byos' => 'ami-09740eee15629d44f',
+            # suse-sles-15-sp2-v20201211-hvm-ssd-x86_64
+            'eu-central-1' => 'ami-09e8a19c9eda495b3'
+        };
+
+        my $ami_id_key = $self->region;
+        $ami_id_key .= '-byos'                     if $img_name =~ /byos/i;
```

Comment:
Don't we need the target `ARCH` here too. If I remember correctly there are some exceptions with ARM and if we have now the possibility to reflect them here, wdyt?
see: https://gitlab.suse.de/qac/qac-openqa-yaml/-/blob/master/sle15/publiccloud/ec2.yaml#L69

Reply:
make sense , will add this